Indicate blockers

willkg opened the pull request:
Fixes #19 

This is slightly different than my original plan. I decided that "loading blockers" was a step of "load bugs", so I made it more like "load comments".

I tweaked the CSS for the needsinfo stuff so it looks a little bit better when there are more than one.

r?

willkg committed:
Indicate blockers

Fixes #19

mythmon commented on edwin/client/static/js/actions/TimelineActions.js:
```diff
@@ -115,9 +121,36 @@ export function loadCommentTags(bugIds) {
+    })
+    .then(() => ProgressActions.endTask('Load blockers'));
+  } else {
+    return undefined;
```
Uh oh. It probably isn't good to sometimes return a Promise and sometimes not. Maybe return something like `Promise.resolve()` (return a promise that is resolved to `undefined`) instead?

mythmon commented on edwin/client/static/js/stores/BugStore.js:
```diff
@@ -211,6 +235,20 @@ BugStore.dispatchToken = Dispatcher.register((action) => {
+        bugMap = bugMap.update(bug.id, new Immutable.Map(), oldBug => {
+          let newBug = Immutable.fromJS(bug);
+          return oldBug.merge(newBug);
+        });
```
I think these 4 lines could be replaced with `bugMap = bugMap.mergeIn([bug.id], Immutable.fromJS(bug))`;

I think.

mythmon commented:
That's... hideous. But it works, so that's cool.

This will bring bugs that aren't tagged with `edwin-${team}` into the bug list as well, right, if they are blocking bugs already in the list? I think that is probably fine.

willkg commented:
It does add non-team bugs into the bug map. So we should probably have the bug lists that bugMap exposes filter them out. I can look into that.

I'll make the other fixes today, too.

willkg committed:
Fix issues Mike found

willkg committed:
Tag all bugs with the team

This tweaks the bugstore a bit so that it knows about "a lot of bugs"
but the queries all return sets of bugs for a specified team. This fixes
the issue where adding blocker bugs to the bugMap added non-team bugs to
the bugMap.

It also fixes the problem where switching between teams caused you to see
a union of the bug sets.

Fixes #61

willkg commented:
^^^ Fixes the issues you found, plus it fixes issue #61.

mythmon commented:
That should work. Lets go with it.
